src/data/error.rs

```diff
@@ -79,3 +80,136 @@ impl<T, E> Recoverable for Result<T, (E, T)> {
         }
     }
 }
+
+pub enum CompileErrorLevel {
+    Warning,
+    RecoverableError,
+    FatalError,
```

Comment:
I would rather these be called `Syntax` and `Semantic`, I think it better describes what they're for. Additionally, syntax errors aren't necessarily fatal.

Either we should get rid of `FatalError` or we should add a `Lex` variant.